instrumentation/metric/registry_test.go

```diff
@@ -0,0 +1,27 @@
+package metric
+
+import (
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestInMemoryRegistry_ExportAll(t *testing.T) {
+	registry := NewRegistry()
+	gauge := registry.NewGauge("hello")
+	gauge.Add(1)
+
+	gaugeValue := registry.ExportAll()["hello"].(gaugeExport)
+	require.EqualValues(t, gaugeValue.Value, 1)
+}
+
+func TestInMemoryRegistry_ExportAndRotate(t *testing.T) {
+	registry := NewRegistry()
+	gauge := registry.NewGauge("hello")
+	gauge.Add(1)
+
+	gaugeValue := registry.ExportAndRotate()["hello"].(gaugeExport)
+	require.EqualValues(t, gaugeValue.Value, 1)
+
+	gaugeValue = registry.ExportAll()["hello"].(gaugeExport)
+	require.EqualValues(t, gaugeValue.Value, 0)
```

Comment:
you could also read the value strait from the `gauge`, no? do you really need to call `ExportAll`?

Reply:
also maybe add a test per metric for the `Reset` operation (if it stays separate from `Rotate`) or for `ExportAndMaybeRotate` if we end up having an atomic operation